Makefile

```diff
@@ -1,73 +1,69 @@
-include config.mk
-
-CFLAGS += -I. -DWLR_USE_UNSTABLE -std=c99 -pedantic -DVERSION=\"$(VERSION)\"
-
-WAYLAND_PROTOCOLS=$(shell pkg-config --variable=pkgdatadir wayland-protocols)
-WAYLAND_SCANNER=$(shell pkg-config --variable=wayland_scanner wayland-scanner)
-
-PKGS = wlroots wayland-server xcb xkbcommon libinput
-CFLAGS += $(foreach p,$(PKGS),$(shell pkg-config --cflags $(p)))
-LDLIBS += $(foreach p,$(PKGS),$(shell pkg-config --libs $(p)))
-
-all: dwl
+.POSIX:
+.SUFFIXES:
 
-clean:
-	rm -f dwl *.o *-protocol.h *-protocol.c
+include config.mk
 
-dist: clean
-	mkdir -p dwl-$(VERSION)
-	cp -R LICENSE* Makefile README.md generate-version.sh client.h\
-		config.def.h config.mk protocols dwl.1 dwl.c util.c util.h\
-		dwl-$(VERSION)
-	echo "echo $(VERSION)" > dwl-$(VERSION)/generate-version.sh
-	tar -caf dwl-$(VERSION).tar.gz dwl-$(VERSION)
-	rm -rf dwl-$(VERSION)
+# flags for compiling
+DWLCPPFLAGS = -I. -DWLR_USE_UNSTABLE -DVERSION=\"$(VERSION)\"
 
-install: dwl
-	install -Dm755 dwl $(DESTDIR)$(PREFIX)/bin/dwl
-	install -Dm644 dwl.1 $(DESTDIR)$(MANDIR)/man1/dwl.1
+# Wayland utils
+WAYLAND_PROTOCOLS = `pkg-config --variable=pkgdatadir wayland-protocols`
+WAYLAND_SCANNER   = `pkg-config --variable=wayland_scanner wayland-scanner`
 
-uninstall:
-	rm -f $(DESTDIR)$(PREFIX)/bin/dwl $(DESTDIR)$(MANDIR)/man1/dwl.1
+# CFLAGS / LDFLAGS
+PKGS      = wlroots wayland-server xkbcommon libinput $(XLIBS)
+DWLCFLAGS = `pkg-config --cflags $(PKGS)` $(DWLCPPFLAGS) $(CFLAGS) $(XWAYLAND)
+LDLIBS    = `pkg-config --libs $(PKGS)`
```

Comment:
Is the goal here to be POSIX compliant or be portable? If it's the former, then I don't think `pkg-config` is specified in POSIX.